Sources/SwifterSwift/Foundation/NSRegularExpressionExtensions.swift

```diff
@@ -0,0 +1,163 @@
+//
+//  NSRegularExpressionExtensions.swift
+//  SwifterSwift
+//
+//  Created by Guy Kogus on 09/10/2019.
+//  Copyright © 2019 SwifterSwift
+//
+
+#if canImport(Foundation)
+import Foundation
+
+public extension NSRegularExpression {
+
+    /// Enumerates the string allowing the Block to handle each regular expression match.
+    ///
+    /// - Parameters:
+    ///   - string: The string.
+    ///   - options: The matching options to report. See `NSRegularExpression.MatchingOptions` for the supported values.
+    ///   - range: The range of the string to test.
+    ///   - block: The Block enumerates the matches of the regular expression in the string.
+    ///     The block takes three arguments and returns `Void`:
+    ///   - result:
+    ///     An `NSTextCheckingResult` specifying the match. This result gives the overall matched range via its `range` property, and the range of each individual capture group via its `range(at:)` method. The range {NSNotFound, 0} is returned if one of the capture groups did not participate in this particular match.
+    ///   - flags:
+    ///     The current state of the matching progress. See `NSRegularExpression.MatchingFlags` for the possible values.
+    ///   - stop:
+    ///     A reference to a Boolean value. The Block can set the value to true to stop further processing of the array. The stop argument is an out-only argument. You should only ever set this Boolean to true within the Block.
+    #if os(Linux)
+    func enumerateMatches(in string: String,
+                          options: MatchingOptions = [],
+                          range: Range<String.Index>,
+                          using block: @escaping (_ result: NSTextCheckingResult?, _ flags: MatchingFlags, _ stop: inout Bool) -> Void) {
```

Comment:
Just curiosity, do you know exactly why this is need for linux? CoreFoundation API requires escaping?

Reply:
The travis build was failing without it: https://travis-ci.org/SwifterSwift/SwifterSwift/jobs/595511815
In the Linux implementation of `enumerateMatches(in:options:range:using)`, `block` is `@escaping`, so if I then try to call a non-escaping block inside it the compiler throws an error.

Reply:
Seems like it 🙃

Reply:
Looks like an implementation detail requirement where the have to store the closure in a [object](https://github.com/apple/swift-corelibs-foundation/blob/432d15abe7aecb6698ab4458c709e2ce97412075/Foundation/NSRegularExpression.swift#L179), making it being required to be escaping